Packs/SaviyntEIC/Integrations/SaviyntEICEventCollector/SaviyntEICEventCollector.py

```diff
@@ -0,0 +1,702 @@
+
+    def fetch_events(
+        self,
+        analytics_name: str,
+        time_frame_minutes: int,
+        max_results: int,
+        offset: int | None = None,
+    ) -> dict[str, Any]:
+        """
+        Fetch events for a single Analytics Runtime Control.
+
+        Args:
+            analytics_name: The event type to fetch (for example, `"SIEMAuditLogs"`).
+            time_frame_minutes: The time frame in minutes to fetch events from.
+            max_results: The maximum number of results to request in this call
+                (the server may cap this value, typically at 10,000).
+            offset: The paging offset to start from, if any.
+
+        Returns:
+            dict[str, Any]: The JSON response returned by the API.
+
+        Notes:
+            On authentication failure (401/Forbidden/invalid token), the client
+            refreshes the token under a lock and retries the request once.
+        """
+        body: dict[str, Any] = {
+            "analyticsname": analytics_name,
+            "attributes": {"timeFrame": str(time_frame_minutes)},
+            "max": max_results,
+        }
+        if offset:
+            body["offset"] = str(offset)
+
+        # Log the executing worker/thread to aid concurrent diagnostics
+        worker_name = threading.current_thread().name
+        demisto.debug(
+            f"[Client.fetch_events] worker={worker_name} analytics={analytics_name} "
+            f"time_frame_minutes={time_frame_minutes} max_results={max_results} offset={offset}"
+        )
+
+        try:
+            return self._http_request(
+                method="POST",
+                url_suffix="api/v5/fetchRuntimeControlsDataV2",
+                headers=self._headers,
+                json_data=body,
+            )
+        except Exception as e:
+            # Attempt one retry on auth failure
+            if any(x in str(e) for x in ("401", "Forbidden", "invalid token")):
+                demisto.debug(f"Access token may be invalid/expired. Attempting to refresh and retry fetch. Error: {e}")
+                # Force refresh token and retry under an auth lock to avoid races across threads
+                with self._auth_lock:
+                    self.obtain_token(force_refresh=True)
+                return self._http_request(
+                    method="POST",
+                    url_suffix="api/v5/fetchRuntimeControlsDataV2",
+                    headers=self._headers,
+                    json_data=body,
+                )
+            raise
+
+
+""" HELPER FUNCTIONS """
+
+
+def add_time_to_events(events: list[dict[str, Any]]):
+    """
+    Add the `_time` field to events based on their occurrence timestamp.
+
+    Args:
+        events: The events to enrich in-place.
+
+    Returns:
+        None
+    """
+    for event in events:
+        create_time = arg_to_datetime(arg=event.get("Event Time"))
+        event["_time"] = create_time.strftime(DATE_FORMAT) if create_time else None
```

Comment:
Are we ok with doing formatting transformation on the data here?

Reply:
Yes. Approved in the Demo.